src/Compilers/CSharp/Portable/Binder/Binder_Lookup.cs

```diff
@@ -1495,6 +1497,161 @@ static bool extendedTypeIsOrDerivedFrom(NamedTypeSymbol hiddenContainer, NamedTy
+                    {
+                        // We wouldn't have gathered these members if we didn't have an extended type for them.
+                        throw ExceptionUtilities.Unreachable();
+                    }
+
+                    if (additionalContainerUnderlying.Equals(currentContainerUnderlying, TypeCompareKind.ConsiderEverything))
+                    {
+                        // neither underlying type is more specific
+                    }
+                    else if (IsDerivedType(baseType: additionalContainerUnderlying, derivedType: currentContainerUnderlying,
+                        basesBeingResolved, this.Compilation, ref useSiteInfo))
+                    {
+                        // current symbol is more specific, it "hides" a less specific non-method/non-indexer additional symbol
+                        if (!IsMethodOrIndexer(additionalSymbol))
+                        {
+                            isAdditionalSymbolHidden = true;
+                        }
+                    }
+                    else if (IsDerivedType(baseType: currentContainerUnderlying, derivedType: additionalContainerUnderlying,
+                        basesBeingResolved, this.Compilation, ref useSiteInfo))
+                    {
+                        // additional symbol is more specific, it "hides" a less specific non-method/non-indexer current symbol
+                        if (!IsMethodOrIndexer(currentSymbol))
+                        {
+                            currentSymbols[j] = null;
+                        }
+                    }
+                    else
+                    {
+                        // neither underlying type is more specific
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// After we've removed all the non-method/non-indexer symbols that are "hidden" by more specific members (of any kind),
+        /// if we have a non-method/non-indexer symbol left in a given hierarchy,
+        /// we should remove any method/indexer symbols that it "hides".
+        ///
+        /// Works in conjunction with <see cref="MergeExtensionLookupResultsHidingLessSpecificNonMethods(LookupResult, LookupResult, ConsList{TypeSymbol}, ref CompoundUseSiteInfo{AssemblySymbol})"/>
+        /// </summary>
+        private void CompleteMergeExtensionLookupResultsHidingLessSpecific(LookupResult result, ConsList<TypeSymbol> basesBeingResolved, ref CompoundUseSiteInfo<AssemblySymbol> useSiteInfo)
+        {
+            var symbols = result.Symbols;
+            var count = symbols.Count;
+
+            for (int i = 0; i < count; i++)
+            {
+                var methodSymbol = symbols[i];
+
+                if (methodSymbol is null || !IsMethodOrIndexer(methodSymbol))
+                    continue;
+
+                for (int j = 0; j < count; j++)
+                {
+                    var nonMethodSymbol = symbols[j];
```

Comment:
>nonMethodSymbol [](http://example.com/codeflow?start=24&length=15)

Why do we assume it is not a method. The naming of locals is confusing. It should reflect what the value is, not what we would like it to be, even if the code performs sum checks later.